parquet/src/main/java/org/apache/iceberg/data/parquet/BaseParquetReaders.java

```diff
@@ -291,6 +291,13 @@ public ParquetValueReader<?> struct(
       return createStructReader(types, reorderedFields, expected);
     }
 
+    private int findIdFromStruct(Types.StructType expected, Type field) {
+      if (field.getId() != null) {
+        return field.getId().intValue();
+      }
+      return expected.field(field.getName()).fieldId();
```

Comment:
If the field-ids are null, how can I find the expected field?

Reply:
I also left a comment on the issue itself https://github.com/apache/iceberg/issues/11214#issuecomment-2514411739. I don't think this is a good solution since it hides the problem, as mentioned before, if you rename a struct field, it will break the table. Instead, I would leverage name mapping as suggested on the issue.

Reply:
Thanks for your comments, I will reconsider and try to re-address the issue.

Reply:
Any time, thank you for the PR, but I'm afraid that it will cause issues downstream